### DISCUSSION TITLE
Future of Open Inference Protocol and supported endpoints on KServe

### DISCUSSION THREAD

**Discussion by olekli** [kubeflow/kfserving#3727] [Q&A]
I am trying to fgure out what the future of OIP is. I have read the minutes in the repository. A `generate` endpoint was proposed and was added to the schemas. In v0.12.0 KServe did support this endpoint. In v0.13.0 this endpoint is remove in favor of OpenAI API endpoints.

Here my questions are:
* Where can I find information about plans like this? (If there is any at all.) On which channel could I get a warning regarding API changes like this?
* What does this mean for OIP? The minutes in the repository stop in Feburary 2024. Before that there seems to have been monthly meetings. Is OIP still alive? The fact that KServe itself is rolling back on the `generate` endpoint raises this question for me.
* In the OIP minutes there was mention of adding more task specific endpoints. Is this something that is planned for KServe?

**Comment by yuzisun** [comment]
@olekli Thanks for the good questions! Sorry for the late response as I just saw this. 
- The generate endpoint was implemented as experimental feature before we decided to adopt the OpenAI schema, the generate endpoint was never officially announced. 
- The open inference protocol monthly meeting is still active and now added to the [official community calendar](https://zoom-lfx.platform.linuxfoundation.org/meetings/kserve?view=month), feel free to join the meeting to learn more about the directions! 
- For task specific APIs please check the [recent proposal doc](https://docs.google.com/document/d/1odTMdIFdm01CbRQ6CpLzUIGVppHSoUvJV_zwcX6GuaU/edit#heading=h.cjkqq1p2wmlx) for how we are going to adopt the current OpenAI spec as the foundation of an industry standard. Our goal is to provide a community maintained and supported specification that servers can adopt, advertise and test compatibility against.
